.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,51 @@
+name: PR Build - Unit Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  unit_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gradle/wrapper-validation-action@v1
+      # The first JDK set up gets to be "primary".
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1.4.0
+        with:
+          java-version: 8
+      - name: save JAVA_HOME as JDK8 for later usage
+        run: echo "::set-env name=ORG_GRADLE_PROJECT_jdk8::$JAVA_HOME"
+      # The second JDK is not first in the path but does reset JAVA_HOME
+      - name: Set up JDK 1.7
+        uses: actions/setup-java@v1.4.0
+        with:
+          java-version: 7
+      - name: save JAVA_HOME as JDK7 for later usage
+        run: echo "::set-env name=ORG_GRADLE_PROJECT_jdk7::$JAVA_HOME"
+      # Java is installed; document settings
+      - name: log project settings
+        run: |
+          echo "--- Java versions ---"
+          $ORG_GRADLE_PROJECT_jdk7/bin/java -version
+          $ORG_GRADLE_PROJECT_jdk8/bin/java -version
+      - name: correct JAVA_HOME
+        run: echo "::set-env name=JAVA_HOME::$ORG_GRADLE_PROJECT_jdk8"
+      # Restore the gradle cache
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          # The docs say to use hashfiles, but gradle itself is smart enough to
+          # re-download dependencies if it couldn't resolve them.
+          # Because our gradle cache is _so big_, I think a 90% cache is far
+          # better than a 0% cache.
+          key: ${{ runner.os }}-gradle
+          restore-keys: |
+            ${{ runner.os }}-gradle
+      # Start running the build.
+      - name: run the unit tests
+        run: ./gradlew --console=plain --parallel test -x functional_test:test -PnoInstrumentation --continue
```

Comment:
It looks like this is running both functional tests and unit tests. Why is that?

Reply:
> I am surprised ./gradlew --console=plain --parallel test runs both unit and functional tests. Didn't know that.

Yes - running an task like `test` that isn't qualified with a project runs that task in all the projects it exists.

> So is functional_test with -PnoInstrumentation a way to disable the functional tests? I see its applied to every instrumentation module...I don't get how that disables a functional test.

The parameters break down like this:
1. `-x functional_test:test` is standard gradle syntax for "don't run the `test` task in the `functional_test` project".
2. `-PnoInstrumentation` is something written in _our_ gradle files that says "don't run the `test` task in any project under `instrumentation`". 

These two options are distinct. The latter doesn't relate to functional tests. Does that answer your question?

Reply:
Ok, that clears it up!